android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -284,25 +358,57 @@ public void onError(String info) {
             stateMachine.transitionToState(StreamingStateMachine.ERROR);
             return;
         }
-        // regardless of VideoStreamingParameters are specified or not, we should refer to VideoStreamingCapability.
+        processCapabilitiesWithPendingStart(encrypted, parameters);
+    }
+
+    private void configureGlobalParameters(Context context, Class<? extends SdlRemoteDisplay> remoteDisplayClass, boolean encrypted, List<VideoStreamingRange> listOfStreamingRange) {
+        this.context = new WeakReference<>(context);
+        this.remoteDisplayClass = remoteDisplayClass;
+        this.isEncrypted = encrypted;
+        this.majorProtocolVersion = internalInterface.getProtocolVersion().getMajor();
+        if (listOfStreamingRange != null) {
+            this.listOfStreamingRanges = listOfStreamingRange;
+        }
+    }
+
+
+    private void processCapabilitiesWithPendingStart(boolean encrypted, VideoStreamingParameters parameters) {
+        final VideoStreamingParameters params = (parameters == null) ? new VideoStreamingParameters() : new VideoStreamingParameters(parameters);
         if (majorProtocolVersion >= 5) {
             if (internalInterface.getSystemCapabilityManager() != null) {
-                final VideoStreamingParameters params = ( parameters == null) ? new VideoStreamingParameters() : new VideoStreamingParameters(parameters);
                 internalInterface.getSystemCapabilityManager().getCapability(SystemCapabilityType.VIDEO_STREAMING, new OnSystemCapabilityListener() {
                     @Override
                     public void onCapabilityRetrieved(Object capability) {
-                        params.update((VideoStreamingCapability) capability, vehicleMake);    //Streaming parameters are ready time to stream
-                        startStreaming(params, encrypted);
+                        VideoStreamingCapability castedCapability = ((VideoStreamingCapability) capability);
+                        VideoStreamManager.this.originalCapability = castedCapability;
+
+                        params.update(castedCapability, vehicleMake);    //Streaming parameters are ready time to stream
+                        VideoStreamManager.this.parameters = params;
+
+                        if (listOfStreamingRanges != null) {
+                            // filtering
+                            castedCapability.setAdditionalVideoStreamingCapabilities(
+                                    getSupportedCapabilities(
+                                            listOfStreamingRanges,
+                                            castedCapability.getAdditionalVideoStreamingCapabilities()
+                                    )
+                            );
+                        }
+                        AppCapability appCapability = new AppCapability(AppCapabilityType.VIDEO_STREAMING);
+                        appCapability.setVideoStreamingCapability(castedCapability);
```

Comment:
The proposal was not very clear on what properties should get set on `OnAppCapabilityUpdated.appCapability. videoStreamingCapability` so we have decided we should not set the root `videoStreamingCapability` but only the `additionalVideoStreamingCapabilities` as outlined below:
1. Mobile checks both the root `videoStreamingCapability` and the `additionalVideoStreamingCapabilities` to see which ones it can support. 
2. Mobile sends an `OnAppCapabilityUpdated` with only the `VideoStreamingCapability.additionalVideoStreamingCapabilities` parameter set. The  `additionalVideoStreamingCapabilities` should be set with all the capabilities it can support as determined by step 1.

Reply:
Hello @kostyaBoss, per this comment:
> Mobile checks both the root `videoStreamingCapability` and the `additionalVideoStreamingCapabilities` to see which ones it can support.

We should update `getSupportedCapabilities(List<VideoStreamingRange>, List<VideoStreamingCapability>)` to instead be `getSupportedCapabilities(List<VideoStreamingRange>, VideoStreamingCapability)`

the VideoStreamingCapability that is passed in will contain a VideoStreamingCapability at the root level as well as the list of additionalVideoStreamingCapabilities. We should be comparing the VideoStreamingCapability at the root level against the ranges as well as the list of additionalVideoStreamingCapabilities.

Currently the implementation is only comparing the list of additionalVideoStreamingCapabilities against the ranges.